packages/terra-folder-tree/src/subcomponents/FolderTreeItem.jsx

```diff
@@ -98,27 +98,35 @@ const FolderTreeItem = ({
     cx(
       'folder-tree-item',
       { selected: isSelected },
-      { folder: isFolder },
       theme.className,
     ),
   );
 
+  const handleRadioButtonClick = (event) => {
+    // Stop click propagation to prevent triggering expand/collapse when selecting folders
+    event.stopPropagation();
+
+    onClick();
+  };
+
   return (
     <>
       <li
         className={itemClassNames}
         role="treeitem"
         aria-expanded={isFolder ? isExpanded : null}
         aria-selected={isSelected}
-        onClick={onToggle}
-        onKeyDown={onToggle}
+        onClick={isFolder ? onToggle : onClick}
+        onKeyDown={isFolder ? onToggle : onClick}
       >
         <input
           type="radio"
           checked={isSelected}
           onChange={onClick}
+          onClick={handleRadioButtonClick}
```

Comment:
What would be the difference between onChange and onClick?  Do we need both event handlers?

Reply:
The `onChange` handler was originally used instead of `onClick` to resolve this [React failed proptype warning](https://stackoverflow.com/questions/36715901/reactjs-error-warning). Unfortunately, stopPropagation will not prevent the row click event if the radio button uses onChange since they are different event types, so I had to use an `onClick` instead.

An alternative to addressing the React warning would be to set `readOnly` on the radio, but that seems incorrect to me as well since it's not really a readonly element, so I've left it as `onChange` for now, but I'm open to other solutions if we don't want to have an extra event handler here.

Reply:
Yep, and that's much nicer, updated: [460c1f2](https://github.com/cerner/terra-framework/pull/1937/commits/460c1f253685252c571ae1926cf60d947cb24e8e)